cmd/microservice-fanfare-fluidity-money/main.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2022 Fluidity Money. All rights reserved. Use of this
+// source code is governed by a GPL-style license that can be found in the
+// LICENSE.md file.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net/url"
+
+	"github.com/fluidity-money/fluidity-app/lib/log"
+	"github.com/fluidity-money/fluidity-app/lib/queues/user-actions"
+	"github.com/fluidity-money/fluidity-app/lib/queues/winners"
+	"github.com/fluidity-money/fluidity-app/lib/types/network"
+	"github.com/fluidity-money/fluidity-app/lib/util"
+	"github.com/fluidity-money/fluidity-app/lib/web"
+	"github.com/fluidity-money/fluidity-app/lib/web/websocket"
+)
+
+// EnvNetwork to match the endpoint for
+const EnvNetwork = `FLU_ETHEREUM_NETWORK`
+
+const (
+	NotificationTypeOnchain = iota + 1
+	NotificationTypeWinningReward
+	NotificationTypeSend
+	NotificationTypeSwapIn
+	NotificationTypeSwapOut
+)
+
+// websocketNotification is received via websocket
+type websocketNotification struct {
+	Type            int    `json:"type"` // NotificationType
+	Source          string `json:"source"`
+	Destination     string `json:"destination"`
+	Amount          string `json:"amount"`
+	Token           string `json:"token"`
+	TransactionHash string `json:"transaction_hash"`
+	RewardType      string `json:"reward_type"`
+}
+
+type registration struct {
+	address string
+	reply   chan *websocket.Broadcast
+}
+
+func main() {
+	network__ := util.GetEnvOrFatal(EnvNetwork)
+
+	network_, err := network.ParseEthereumNetwork(network__)
+
+	if err != nil {
+		log.Fatal(func(k *log.Log) {
+			k.Message = "Failed to parse the Ethereum network!"
+			k.Payload = err
+		})
+	}
+
+	endpoint, err := url.JoinPath("/", string(network_))
+
+	if err != nil {
+		log.Fatal(func(k *log.Log) {
+			k.Message = "Failed to join the network to create an endpoint!"
```

Comment:
This is phrased weirdly, I didn't understand what it meant until I read the code. Maybe `Failed to create network endpoint!`? Also could use the first variable `network__` rather than re-stringifying